### PR TITLE
seboolean: Fix issue with changing persistent SELinux boolean values

### PR DESCRIPTION
**Pull request by mkrupcale**
##### SUMMARY
Previously, when the active SELinux boolean value was the same as the desired value, regardless of the value of `persistent`, the module would simply exit successfully, assuming no need for update. This made the assumption that the active and persistent values should be the same, but that is not always the case. This modification treats both the active and persistent values separately when checking for the need to update. Note that the persistent update mode, however, will still update the active as well as the persistent value. If this is not desired, it is possible to separately toggle the active value alone.

Other changes:
* Make the check mode actually perform checks for changes
* Organizes semanage commands into set of logical steps

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
seboolean

##### ANSIBLE VERSION
```
ansible 2.4.0 (seboolean-persistent-changed 94283d29d9) last updated 2017/03/19 22:20:16 (GMT -400)
  config file = 
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jan 12 2017, 17:59:37) [GCC 6.3.1 20161221 (Red Hat 6.3.1-1)]
```

##### ADDITIONAL INFORMATION
To test this, we can set the active SELinux boolean value and then change the persistent value to the same value.
```
$ sudo semanage boolean -l | grep use_fusefs_home_dirs
use_fusefs_home_dirs           (off  ,  off)  Allow use to fusefs home dirs
$ ansible localhost -b -K -m seboolean -a 'name="use_fusefs_home_dirs" state="yes" persistent="no"'
$ sudo semanage boolean -l | grep use_fusefs_home_dirs
use_fusefs_home_dirs           (on   ,  off)  Allow use to fusefs home dirs
$ ansible localhost -b -K -m seboolean -a 'name="use_fusefs_home_dirs" state="yes" persistent="yes"'
$ sudo semanage boolean -l | grep use_fusefs_home_dirs
use_fusefs_home_dirs           (on   ,   on)  Allow use to fusefs home dirs
```
Note that in the previous version, the attempt to toggle the persistent value to "on" while the active value was "on" would exit before doing so.

### CODE REVIEW

File: lib/ansible/modules/system/seboolean.py
Comment (bcoca) [inline lib/ansible/modules/system/seboolean.py]: this should note the version on which these requirements were added

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/system/seboolean.py
Comment (mkrupcale) [reply lib/ansible/modules/system/seboolean.py]: It looks like seboolean was first committed in version 0.7 97c2e58081c9c7c139916d8164bd649ba8409c0e, and it has always had these requirements, but when the documentation was added later in 054f99b10de6c092232855b1e6a27d0160020059, it did not include them. What is the proper way to indicate then that version 0.7 is the version these requirements were added? Would that go in the `notes` part?
http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#documentation-block only talks about optionally specifying the minimum version required.

Comment (mkrupcale) [reply lib/ansible/modules/system/seboolean.py]: @bcoca This matter appears to have been resolved in PR #23029, which has since been merged, and I have rebased my changes on top of this.

**Comment by mkrupcale** [comment]
@bcoca I have fixed some merge conflicts so that this is now current. Can this get a second review now?

**Comment by mkrupcale** [comment]
@bcoca Fixed merge conflict after PEP8 changes.

**Comment by mkrupcale** [comment]
@bcoca Fixed merge conflict & ready for review.

**Review by bcoca — APPROVED** [review]

**Comment by bcoca** [comment]
rebuild_merge

**Pull request merged**
